clang/test/Driver/darwin-print-libgcc-file-name.c

```diff
@@ -0,0 +1,97 @@
+// Test the output of -print-libgcc-file-name on Darwin.
```

Comment:
I added these tests and unfortunately they fail (both before and after my change). I believe part of the problem is that `-print-libgcc-file-name` is still only dealing with the default toolchain we construct first. The actual toolchain object used is constructed much much later, so early initialization is not sufficient to fix this particular problem. I added these tests as expected to fail to document this shortcoming. I still believe we should take this change as it at least fixes the behavior of this flag for the non-embedded scenarios. 

The proper fix would be really involved. I shortly experimented with moving  the processing of `-print-libgcc-file-name` to a later stage in the driver, but that is also problematic as clang might start to emit errors about missing input files in that case (because having an input file is not a requirement for this particular flag, but it is a requirement to build the jobs/actions that happen before we construct the actual toolchain that has the correct data we need). 

I am not really sure what is the correct solution here, I start to think that the `-print-libgcc-file-name` flag is just not compatible with the way clang works, and we might need a separate flag that needs input files as well and is processed at a later stage and is more reliable for all targets.